src/test/suite/oidc.test.ts

```diff
@@ -398,6 +406,7 @@ suite('OIDC Tests', function () {
     }
 
     await reAuthPromise;
+    await sleep(100);
```

Comment:
Why do we need this sleep now? Can this be down without it?

Reply:
I wanted to ensure here that it is enough time to capture the closing event. Let's try without sleeps, and see how tests react.

Reply:
Yeah we do need sleep here otherwise the connection Is not yet closed, but I cleaned up some other places that you have mentioned.

Reply:
Instead of `sleep` we can use something like a `waitFor` with a timeout that keeps increasing, that way we will complete in 5ms if the end condition is already met.